deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/graph/vertex/impl/ElementWiseVertex.java

```diff
@@ -125,8 +149,27 @@ public INDArray doForward(boolean training) {
                     }
                 }
                 return new Pair<>(null, out_product);
+            case Max:
+                INDArray[] outMax = new INDArray[nInForwardPass];
+                INDArray maxIndices = epsilon.dup();
+                CustomOp op = DynamicCustomOp.builder("mergemaxindex")
+                        .setInputs(inputs)
+                        .setOutputs(maxIndices)
+                        .callInplace(false)
+                        .build();
+                Nd4j.getExecutioner().exec(op);
+                for (int i = 0; i < nInForwardPass; i++) {
+                    //gradient is epsilon where the max index is the same as i and zero elsewhere
+                    outMax[i] = maxIndices.dup();
+                    //generate a mask with 1s and 0s in the right places
+                    Nd4j.getExecutioner().exec(new CompareAndSet(outMax[i], -1 , Conditions.notEquals(i)));
+                    Nd4j.getExecutioner().exec(new CompareAndSet(outMax[i], 1 , Conditions.notEquals(-1)));
+                    Nd4j.getExecutioner().exec(new CompareAndSet(outMax[i], 0 , Conditions.equals(-1)));
```

Comment:
I can't think of a better way - but 3 passes isn't efficient. Maybe we add a version of CompareAndSet that basically does
```x = (condition(x) ? a : b``` instead of the current ```x = (condition(x) ? a : x```. That would make this doable in 1 pass.

Reply:
Lets just add new op. It takes just a few minutes.

Reply:
@raver119 Sounds great. Thanks